kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/richmessaging/FaqAlRichMessage.java

```diff
@@ -0,0 +1,89 @@
+package com.applozic.mobicomkit.uiwidgets.conversation.richmessaging;
+
+import android.content.Context;
+import android.text.TextUtils;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import com.applozic.mobicomkit.api.conversation.Message;
+import com.applozic.mobicomkit.uiwidgets.AlCustomizationSettings;
+import com.applozic.mobicomkit.uiwidgets.R;
+import com.applozic.mobicomkit.uiwidgets.conversation.richmessaging.callbacks.ALRichMessageListener;
+import com.applozic.mobicomkit.uiwidgets.conversation.richmessaging.models.ALRichMessageModel;
+import com.applozic.mobicommons.json.GsonUtils;
+
+import java.util.List;
+
+public class FaqAlRichMessage extends AlRichMessage {
+
+    public FaqAlRichMessage(Context context, LinearLayout containerView, Message message, ALRichMessageListener listener, AlCustomizationSettings alCustomizationSettings) {
+        super(context, containerView, message, listener, alCustomizationSettings);
+    }
+
+    @Override
+    public void createRichMessage() {
+        super.createRichMessage();
+        LinearLayout faqReplyLayout = containerView.findViewById(R.id.alFaqReplyLayout);
+        LinearLayout faqLayout = containerView.findViewById(R.id.alFaqLayout);
+        setupAlRichMessage(faqLayout, faqReplyLayout, model);
+    }
+
+    //setup he actionYes and actionNo text views
+    void setActionTextView(TextView actionTextView, int index, List<ALRichMessageModel.AlButtonModel> actionModel, ALRichMessageModel.ALPayloadModel payload) {
```

Comment:
actionModel and index are not getting used exclusively inside this function, what we are using inside it is always a combination of both i.e.
actionModel.get(index) 
so we can remove both the parameter and pass a single parameter with value as actionModel.get(index)

Reply:
done